test/e2e/scheduling/ubernetes_lite.go

```diff
@@ -52,19 +55,30 @@ var _ = SIGDescribe("Multi-AZ Clusters", func() {
 		msg := fmt.Sprintf("Zone count is %d, only run for multi-zone clusters, skipping test", zoneCount)
 		e2eskipper.SkipUnlessAtLeast(zoneCount, 2, msg)
 		// TODO: SkipUnlessDefaultScheduler() // Non-default schedulers might not spread
+
+		cs = f.ClientSet
+		nodeList = &v1.NodeList{}
+		e2enode.WaitForTotalHealthy(cs, time.Minute)
+		nodeList, err = e2enode.GetReadySchedulableNodes(cs)
+		framework.ExpectNoError(err)
 	})
 	ginkgo.It("should spread the pods of a service across zones", func() {
-		SpreadServiceOrFail(f, 5*zoneCount, imageutils.GetPauseImageName())
+		SpreadServiceOrFail(f, 5*zoneCount, imageutils.GetPauseImageName(), nodeList)
 	})
 
 	ginkgo.It("should spread the pods of a replication controller across zones", func() {
-		SpreadRCOrFail(f, int32(5*zoneCount), framework.ServeHostnameImage, []string{"serve-hostname"})
+		SpreadRCOrFail(f, int32(5*zoneCount), framework.ServeHostnameImage, []string{"serve-hostname"}, nodeList)
 	})
 })
 
 // SpreadServiceOrFail check that the pods comprising a service
 // get spread evenly across available zones
-func SpreadServiceOrFail(f *framework.Framework, replicaCount int, image string) {
+func SpreadServiceOrFail(f *framework.Framework, replicaCount int, image string, nodeList *v1.NodeList) {
+	// make the nodes have balanced cpu,mem usage
+	cleanUp, err := createBalancedPodForNodes(f, f.ClientSet, f.Namespace.Name, nodeList.Items, podRequestedResource, 0.6)
```

Comment:
why not put this in the SIGDescribe? That is, just do it once for the 2 tests.

Reply:
I think each test has its own namespace created by the e2e/ginkgo framework, which includes built in cleanup of that namespace ([looking here](https://github.com/kubernetes/kubernetes/blob/3416b6b0a4e7b76cb38f6b1749766599b3fd2368/test/e2e/framework/framework.go#L162)). 

If that's the case, then we want to create the balancing pods in each test's namespace, otherwise they will get cleaned up after the first test.

Reply:
we are sharing the framework instance, so I don't think so.

Reply:
I mean, that the framework instance includes [`BeforeEach`/`AfterEach`](https://github.com/kubernetes/kubernetes/blob/3416b6b0a4e7b76cb38f6b1749766599b3fd2368/test/e2e/framework/framework.go#L185-L186) functions that [update the value of `f.Namespace`](https://github.com/kubernetes/kubernetes/blob/3416b6b0a4e7b76cb38f6b1749766599b3fd2368/test/e2e/framework/framework.go#L249) and [clear that value](https://github.com/kubernetes/kubernetes/blob/3416b6b0a4e7b76cb38f6b1749766599b3fd2368/test/e2e/framework/framework.go#L418), respectively. I could manually run it to double check

Reply:
oh you are right. The other thing you can do is call the clean function in `AfterEach`

Reply:
Similar to what we talked about above (https://github.com/kubernetes/kubernetes/pull/98699#discussion_r569586155), we generate the `cleanUp` function by passing the framework namespace to `createBalancedPods`, and immediately defer the call to the end of the current (test) function. So there is not really a way to get that function in `AfterEach`

For reference, this is the same pattern we have been using in the scheduler priorities serial tests: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduling/priorities.go#L125

Reply:
instead of directly using defer, you just call the cleanup function from `AfterEach`

```golang
var cleanup func()
ginko.BeforeEach(func() {
  cleanup = setup()
}
ginko.AfterEach(func() {
  if cleanup != nil {
    cleanup()
  }
}
```

Reply:
Ah, sure that works too. Updated